packages/ui/VirtualList/VirtualListBase.js

```diff
@@ -673,6 +833,69 @@ const VirtualListBaseFactory = (type) => {
 			}
 		}
 
+		// For VariableVirtualList
+		adjustVariableGridPosition () {
```

Comment:
I'm concerned this method is doing too much work. Appears that it is:

* calculating positions and caching
* applying positions to nodes
* updating the thresholds
* updating scroll bounds
* setting container size
* updating the moreInfo struct

We should at least decompose these a bit into separate functions for better readability and determine which should really be combined into a single method vs which should be called individually from other places.

Reply:
First of all, I decomposed these into separate functions. But I hesitated to refactor these more because I didn't know how much I had to do more from now. Please let me know your thoughts.

Reply:
Looks much better! We can continue to refine as needed but you made some great progress over night. Thanks!!